app/src/org/commcare/CommCareApplication.java

```diff
@@ -301,22 +295,9 @@ public SecretKey createNewSymmetricKey() {
         return getSession().createNewSymmetricKey();
     }
 
-    synchronized public Tracker getDefaultTracker() {
-        if (analyticsTracker == null) {
-            if (BuildConfig.DEBUG) {
-                analyticsTracker = analyticsInstance.newTracker(DEV_TRACKING_ID);
-            } else {
-                analyticsTracker = analyticsInstance.newTracker(LIVE_TRACKING_ID);
-            }
-            analyticsTracker.enableAutoActivityTracking(true);
-        }
-        String userId = getCurrentUserId();
-        if (!"".equals(userId)) {
-            analyticsTracker.set("&uid", userId);
-        } else {
-            analyticsTracker.set("&uid", null);
-        }
-        return analyticsTracker;
+    synchronized public FirebaseAnalytics getAnalyticsInstance() {
+        analyticsInstance.setUserId(getUserIdOrNull());

```

Comment:
Will this throw an NPE if `FirebaseAnalyticsUtil.versionIncompatible` is true on line 218?

Reply:
Yea good point. Technically this is fine right now because all calls to `getAnalyticsInstance()` are protected by a version compatibility check first, but this definitely isn't the best way to do it. Will make a change to the control flow